downloader.go

```diff
@@ -317,16 +319,22 @@ func DefaultDownloader() *Downloader {
 		MaxRetries:           DefaultMaxRetries,
 		ChunkSize:            DefaultChunkSize,
 		WaitRetry:            DefaultWaitRetry,
-		client:               newClient(DefaultMaxRetries, DefaultTimeout),
+		Client:               NewHTTPClient(nil, DefaultMaxRetries, DefaultTimeout),
 	}
 }
 
-func newClient(maxParallelDownloadsPerServer int, timeoutPerChunk time.Duration) *http.Client {
+// NewHTTPClient wraps the logic of using timeout and HTTP transport to control
+// the HTTP requests used by Chunk. It can create a new HTTP client if none
+// is passed, or can overwrite these settings in an existing one. It is helper
+// function for use cases in which a customized HTTP client is needed.
+func NewHTTPClient(c *http.Client, maxParallelDownloadsPerServer int, timeoutPerChunk time.Duration) *http.Client {
```

Comment:
Sorry, but I believe this way, we are making the usage harder, @cuducos (I understand it is well documented). The main reason for the confusion is that we are allowing some degrees of freedom by accepting the HTTP client and forcefully overriding its transport with our configuration. So, the user can not configure other transport parameters, for example, disable compression, TLS, or proxying.
 
That said, one way to improve this clarity is:

1. Undo exporting `newHTTPClient` (essentially leaving it as it was)
2. Export client

With that, the user could do:

```go
d := chunk.DefaultDownloader()
d.Client = &http.Client{ ... }
d.Download(...)
```

What do you think?

Reply:
This is already possible with the code proposed in this PR as it is. Lines 61-65 export the HTTP client and document it. So I totally understand the use case you have in mind : )

The idea of `NewHTTPClient` was to apply our non-trivial settings to it, _in case the user wants it_. But it's just a helper, users can just assign any client to the struct as it is.

Reply:
What about renaming it to something like `ApplyTimoutAndTransportSettings(…)` and keeping it around as a helper for users that want to customize parts of it, but want to trust us to do these specific settings?

Reply:
> The idea of NewHTTPClient was to apply our non-trivial settings to it, in case the user wants it. But it's just a helper, users can just assign any client to the struct as it is.

Isn't it simpler for the user to understand just getting the `DefaultDownloader()` and then overriding whatever it needs? Both reach the goal, but I don't get the reason for adding this extra complexity. Sorry if I am missing something obvious. Could you please explain?

> What about renaming it to something like ApplyTimoutAndTransportSettings(…) and keeping it around as a helper for users that want to customize parts of it, but want to trust us to do these specific settings?

The same question above applies. To me, it is really just a matter of getting the default settings and then overriding whatever they want.